app/models/task.rb

```diff
@@ -62,34 +53,23 @@ def startable?
     task_status == 'unstarted'
   end
 
-  def self.no_started? task
-    parents = Taskedge.find_parents task
-    parents.each do |parent|
-      if Task.find(parent).task_status != 'finished'
-        return true
-      end
-    end
-    false
-  end
-
-  def self.add_taskedge(parent_id, child_id)
-    if Task.exists?(parent_id) and Task.exists?(child_id)
-      edge = Taskedge.new
-      edge.parenttask_id = parent_id
-      edge.childtask_id = child_id
-      edge.save
-    else
-      nil
-    end
+  # create a task in iteration or template through params
+  def self.create_task params
```

Comment:
I think you can pass params in the constructor.  `Task.new params` should work.  Although you may want to validate and sanitize the params.

Reply:
fixed by directly call create